### PR TITLE
Fixed https://github.com/couchbase/couchbase-lite-java-core/issues/1255

### PR DESCRIPTION
**Pull request by hideki**
Support OpenID connect
- Add TokenStore for Android with using KeyStore functionality
- Implemented Unit Test for OpenID Connect Authenticator

### CODE REVIEW

File: src/androidTest/assets/test.properties [hunk 1]
```diff
@@ -2,8 +2,10 @@
 # copy this file to "local-test.properties" in the same directory
 # values in that file will override these
 # and it will be ignored by git automatically
-replicationUrl=http://10.0.2.2:4984/db
-adminReplicationUrl=http://10.0.2.2:4985/db
+replicationUrl=http://10.0.0.9:4984/db
+adminReplicationUrl=http://10.0.0.9:4985/db
```
Comment (pasin) [inline src/androidTest/assets/test.properties]: What is 10.0.0.0 IP Address? Is this the new Android Emulator IP?

File: src/main/java/com/couchbase/lite/auth/TokenStoreFactory.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+//
+package com.couchbase.lite.auth;
+
+import android.content.Context;
+import android.os.Build;
+
+/**
+ * Created by hideki on 6/22/16.
+ */
+public class TokenStoreFactory {
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/auth/TokenStoreFactory.java]: Can you explain how TokenStoreFactory used for users?

File: src/main/java/com/couchbase/lite/auth/TokenStore.java [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+/**
+ * Created by hideki on 6/22/16.
+ */
+public class TokenStore implements ITokenStore {
+
+    public static final String TAG = Log.TAG_SYNC;
+
+    // https://developer.android.com/training/articles/keystore.html#SupportedCiphers
+    private static final String CIPHER_ALGORITHM = "RSA/ECB/PKCS1Padding";
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/auth/TokenStore.java]: Can you double check if we are using the same algorithm as the database encryption?

Comment (hideki) [inline src/main/java/com/couchbase/lite/auth/TokenStore.java]: `AES` is not supported by API 18, it requires API 23 +.
https://developer.android.com/training/articles/keystore.html#SupportedCiphers

File: src/main/java/com/couchbase/lite/auth/TokenStoreFactory.java (hunk 2 above)
Comment (hideki) [inline src/main/java/com/couchbase/lite/auth/TokenStoreFactory.java]: To create OpenIDConnectAuthenticator, it requires TokenStore in addition to callback.
```
public static Authenticator createOpenIDConnectAuithenticator(OpenIDConnectAuthorizer.OIDCLoginCallback callback, ITokenStore tokenStore) {
   return new OpenIDConnectAuthorizer(callback, tokenStore);
}
```
https://github.com/couchbase/couchbase-lite-java-core/pull/1284/files#diff-b8325d17333d62df13ec5bd23e2b7820R59

maybe we could overrides `AuthenticatorFactory` for Android.

**Pull request merged**